airflow/bin/cli.py

```diff
@@ -165,6 +171,29 @@ def run(args):
         executor.heartbeat()
         executor.end()
 
+    if conf.get('core', 'S3_LOG_FOLDER').startswith('s3:'):
+        import boto
+        s3 = boto.connect_s3()
+        s3_log = filename.replace(log, conf.get('core', 'S3_LOG_FOLDER'))
+        bucket, key = s3_log.lstrip('s3:/').split('/', 1)
+        s3_key = boto.s3.key.Key(s3.get_bucket(bucket), key)
+        if os.path.exists(filename):
+
+            # get logs
+            with open(filename, 'r') as logfile:
+                new_log = logfile.read()
+
+            # remove old logs (since they are already in S3)
+            new_log.replace(old_log, '')
+
+            # append new logs to old S3 logs, if available
+            if s3_key.exists():
+                old_s3_log = s3_key.get_contents_as_string().decode()
+                new_log = old_s3_log + '\n' + new_log
+
+            # send log to S3
+            s3_key.set_contents_from_string(new_log)
```

Comment:
I'm having an internal debate on wether we should wrap this in a try statement. An s3 outage would make everything fails, but if s3 goes down, probably a significant portion of modern computing is probably down too...